exp/addons/controllers/clusterresourceset_controller_test.go

```diff
@@ -292,21 +292,18 @@ metadata:
 			},
 			Data: map[string]string{},
 		}
+
+		// Let's wait until the initial reconciliations are *all* done, so we can test
+		// if the ConfigMap creation triggers a reconciliation. Otherwise a previous change
+		// of another resource triggers the reconciliation and we cannot verify if we react
+		// correctly to the ConfigMap creation
+		time.Sleep(5 * time.Second)
+
 		Expect(testEnv.Create(ctx, newConfigmap)).To(Succeed())
 		defer func() {
 			Expect(testEnv.Delete(ctx, newConfigmap)).To(Succeed())
 		}()
 
-		cmKey := client.ObjectKey{
```

Comment:
I'm not sure what the purpose of the Get was. It should just return instantly because the Create above is synchronous. Maybe it was a try to ensure that the ConfigMap exists in the local cache to fix some test flakiness?

Reply:
> Maybe it was a try to ensure that the ConfigMap exists in the local cache to fix some test flakiness?

This sounds about right 😅  All the controller runtime stuff uses cached clients normally so if `testEnv` is also caching, this would be it waiting until the informer has received the create event and updated the cache. Looking at the next `Eventually`, this basically allows us to know that the timeout is used solely for that reconcile. 

Have you tried running this many times to see if you can get it to fail without this extra block?

Reply:
Not yet, I'll try. Although the 10s timeout is easily enough. So it's highly likely that there won't be any difference.

Reply:
I ran it locally in a loop a few hundred times. But I think we don't need the `Get`. The `Eventually` below tries for 20 seconds and locally it takes `< 1s`. But I don't really care, if the consensus is that we should keep it, I'll just add it again.

Reply:
Let's add it back, having a `Get` to make sure the cache syncs is the most idiomatic way to deal with caches and controllers lagging behind a bit